src/util/sharing_map.h

```diff
@@ -887,105 +893,46 @@ ::erase_all(const keyst &ks, const tvt &key_exists)
   return cnt;
 }
 
-/// Insert element, return const reference
+/// Insert element, element must not exist in map
 ///
 /// Complexity:
 /// - Worst case: O(H * S + M)
 /// - Best case: O(H)
 ///
 /// \param k: The key of the element to insert
 /// \param m: The mapped value to insert
-/// \param key_exists: Hint to indicate whether the element is known to exist
-///   (possible values `false` or `unknown`)
-/// \return Pair of const reference to existing or newly inserted element, and
-///   boolean indicating if new element was inserted
-SHARING_MAPT2(const, const_find_type)
-::insert(const key_type &k, const mapped_type &m, const tvt &key_exists)
+SHARING_MAPT4(U, void)
```

Comment:
Except for a single `<T>`, I think you should avoid single-character template parameter names. This one looks like ValueT perhaps?

Reply:
Yes this is the type for the use in the forwarding reference for the value. I'll rename it to `valueU` (`valueT` is already taken here).